cmd/bosun/expr/funcs.go

```diff
@@ -528,6 +570,120 @@ func GraphiteBand(e *State, T miniprofiler.Timer, query, duration, period, forma
 	}
 	return
 }
+
+func Window(e *State, T miniprofiler.Timer, query, duration, period string, num float64, rfunc string) (r *Results, err error) {
+	fn, ok := e.GetFunction(rfunc)
+	if !ok {
+		return nil, fmt.Errorf("expr: Window: no %v function", rfunc)
+	}
+	f := reflect.ValueOf(fn.F)
+	r = new(Results)
+	r.IgnoreOtherUnjoined = true
+	r.IgnoreUnjoined = true
+	T.Step("window", func(T miniprofiler.Timer) {
+		var d, p opentsdb.Duration
```

Comment:
This whole first part should probably be a function since both Band and Window call it.